File: app/src/main/java/com/king/frame/base/BaseFragment.java
package com.king.frame.base;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.text.TextUtils;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

/**
 * 基础Fragment
 */
public abstract class BaseFragment extends Fragment {

    protected View mContentView;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        mContentView = inflater.inflate(getContentView(), container, false);
        initView();
        loadData();
        return mContentView;
    }

    protected abstract int getContentView();

    protected abstract void initView();

    protected abstract void loadData();


    public void showShortToast(String msg) {
        if (!TextUtils.isEmpty(msg)) {
            Toast.makeText(getActivity(), msg, Toast.LENGTH_SHORT).show();
        }
    }

}

File: app/src/main/java/com/king/frame/base/BaseFragment.kt
package com.king.frame.base

import android.os.Bundle
import android.text.TextUtils
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Toast

import androidx.fragment.app.Fragment

/**
 * 基础Fragment
 *
 * @author king
 * @date 2019-11-27 14:30
 */
abstract class BaseFragment : Fragment() {

    protected var mContentView: View? = null

    protected abstract val contentView: Int

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        mContentView = inflater.inflate(contentView, container, false)
        initView()
        return mContentView
    }

    protected abstract fun initView()



    fun showShortToast(msg: String) {
        if (!TextUtils.isEmpty(msg)) {
            Toast.makeText(activity!!, msg, Toast.LENGTH_SHORT).show()
        }
    }

}
